lib/select-builder.js

```diff
@@ -1,89 +1,74 @@
 'use strict';
 
-const { iter } = require('@metarhia/common');
-
 const { QueryBuilder } = require('./query-builder');
 
-const allowedConditions = new Set(['=', '<>', '<', '<=', '>', '>=', 'LIKE']);
-
-const supportedOps = {
-  select: Set,
-  innerJoin: Array,
-  selectDistinct: null,
-  count: Array,
-  avg: Array,
-  min: Array,
-  max: Array,
-  sum: Array,
-  where: Array,
-  groupBy: Set,
-  orderBy: Set,
-  from: Set,
-  limit: null,
-  offset: null,
-};
+const compareOperators = ['=', '<>', '<', '<=', '>', '>=', 'LIKE'];
+const setOperators = ['= ANY', 'IN', 'NOT IN'];
+const operators = [...compareOperators, ...setOperators];
 
-const functionHandlers = {
-  count: op => `count(${op.field})`,
-  avg: op => `avg(${op.field})`,
-  min: op => `min(${op.field})`,
-  max: op => `max(${op.field})`,
-  sum: op => `sum(${op.field})`,
-};
+const sqlFunctions = ['count', 'avg', 'min', 'max', 'sum'];
 
-const parseCondition = cond => {
-  if (cond === '!=') return '<>';
-  if (!allowedConditions.has(cond)) {
-    throw new Error(`The operator "${cond}" is not permitted`);
+const parseOperator = op => {
+  if (op === '!=') return '<>';
+  if (!operators.includes(op)) {
+    throw new Error(`The operator "${op}" is not permitted`);
   }
-  return cond;
+  return op;
 };
 
 const checkType = (value, name, type) => {
   if (!(value instanceof QueryBuilder) && typeof value !== type) {
-    throw new TypeError(
-      `Invalid '${name}' value (${value}) type, expected '${type}'`
-    );
+    const msg = `Invalid '${name}' value (${value}) type, expected '${type}'`;
+    throw new TypeError(msg);
   }
 };
 
-const makeParamValue = (cond, value, params) => {
-  if (value instanceof QueryBuilder) {
-    return '(' + value.build(params) + ')';
-  } else if (cond === 'IN' || cond === 'NOT IN') {
+const makeParamValue = (op, value, params) => {
+  if (value instanceof QueryBuilder) return '(' + value.build(params) + ')';
+  if (op === 'IN' || op === 'NOT IN') {
     return '(' + value.map(v => params.add(v)).join(', ') + ')';
-  } else if (cond && cond.endsWith('ANY')) {
-    return '(' + params.add(value) + ')';
-  } else {
-    return params.add(value);
   }
+  const par = params.add(value);
+  if (op && op.endsWith('ANY')) return '(' + par + ')';
+  return par;
```

Comment:
IMO the previous version was clearer, each branch handled one use case =).

Reply:
Also rewritten to use concatenate